sdk/javascript/generator/StructTypeFormatter.py

```diff
@@ -0,0 +1,366 @@
+
+						# assume same size and generate single dummy access
+						comment = '// deserialize to temporary buffer for further processing'
+						deserialize = f'const {field.extensions.printer.name}Temporary = {field.extensions.printer.load()};'
+						temporary_buffer = create_temporary_buffer_name(condition_field_name)
+						temporary_buffer_limit = f'{field.extensions.printer.name}Temporary.size'
+						temporary = f'const {temporary_buffer} = view.window({temporary_buffer_limit});'
+						adjust = f'view.shiftRight({temporary_buffer_limit}); // skip temporary'
+						body += comment + '\n' + deserialize + '\n' + temporary + '\n' + adjust + '\n\n'
+
+					# queue field for re-reading it from temporary buffer
+					queued_fields[condition_field_name].append({'field': field})
+					continue
+
+			deserialized_field = self.generate_deserialize_field(field)
+			body += deserialized_field
+			processed_fields.add(field.name)
+
+			# if conditional field has been processed, process queued fields
+			for conditioned in queued_fields.get(field.name, []):
+				body += self.generate_deserialize_field(
+					conditioned['field'],
+					create_temporary_buffer_name(field.name),
+				)
+
+		# create call to ctor
+		body += '\n'
+		body += f'const instance = new {self.typename}();\n'
+
+		for field in self.non_reserved_fields():
+			field_name = self.field_name(field, 'instance')
+			body += f'{field_name} = {field.extensions.printer.name};\n'
+
+		body += 'return instance;'
+		return MethodDescriptor(body=body)
+
+	def generate_serialize_field(self, field):
+		condition = self.generate_condition(field, True)
+
+		field_value = ''
+		field_comment = ''
+
+		# bound fields are the size / count / sizeof fields that are bound to either object or array
+		bound_field = field.extensions.bound_field
+		if is_bound_size(field):
+			bound_field_name = self.field_name(bound_field)
+			field_comment = f' // bound: {field.name}'
+
+			if bound_field.display_type.is_array:
+				if field.name.endswith('_count') or not bound_field.field_type.is_byte_constrained:
+					field_value = f'{bound_field_name}.length'
+
+					bound_condition = self.generate_condition(bound_field, True)
+					if condition and bound_condition:
+						raise RuntimeError('do not know yet how to generate both conditions')
+
+					# HACK: create inline if condition (for NEM namespace purposes)
+					if bound_condition:
+						condition_value = bound_field.value.value
+						field_value = f'({bound_field_name} ? {field_value} : {condition_value})'
+				else:
+					field_value = bound_field.extensions.printer.get_size()
+			elif field.is_size_reference:
+				field_value = bound_field.extensions.printer.get_size()
+		else:
+			field_value = self.field_name(field)
+
+		serialize_line = None
+		if DisplayType.TYPED_ARRAY == field.display_type:
+			serialize_field = field.extensions.printer.store(field_value, 'buffer_')
+			serialize_line = f'{serialize_field};{field_comment}\n'
```

Comment:
hack to generate something like:
`arrayHelpers.writeArray(buffer_, this._mosaics, e => e.mosaicId.value);`

larger sample:
```js
		buffer_.write(converter.intToBytes(this._message.length, 2, false)); // bound: message_size
		buffer_.write(converter.intToBytes(this._mosaics.length, 1, false)); // bound: mosaics_count
		buffer_.write(converter.intToBytes(this._transferTransactionBodyReserved_1, 1, false));
		buffer_.write(converter.intToBytes(this._transferTransactionBodyReserved_2, 4, false));
		arrayHelpers.writeArray(buffer_, this._mosaics, e => e.mosaicId.value);
		buffer_.write(this._message);
```